bioio/bio_image.py

```diff
@@ -255,6 +255,28 @@ def determine_plugin(
             ),
         )
 
+    @staticmethod
+    def _path_has_extension(path: str, extension: str) -> bool:
+        """
+        Examples
+        --------
+        >>> BioImage._path_has_extension("https://example.com/file.czi", ".czi")
+        True
+        >>> BioImage._path_has_extension("https://example.com/file.czi?x=123", ".czi")
+        True
+        >>> BioImage._path_has_extension("./file.ome.zarr", ".ome.zarr")
+        True
+        >>> BioImage._path_has_extension("./file.ome.zarr", ".zarr")
+        True
+        >>> BioImage._path_has_extension("./file.ome.zarr", ".ome")
+        False
+        """
+
+        def ends_with_ext(string: str) -> bool:
+            return string.lower().endswith(extension)
+
+        return ends_with_ext(path) or ends_with_ext(path.split("?")[0])
```

Comment:
Is this the only case we are handling? Is this just for S3 files?

Reply:
This is the only practical use case I know of at this time. It seems plausible that other web servers hosting CZI files might use query strings we'd like to ignore.

Reply:
Nice! this looks good to me then. like the sub functions